cli/commands/site/list.ts

```diff
@@ -1,69 +1,98 @@
+import os from 'node:os';
 import { __, _n, sprintf } from '@wordpress/i18n';
 import Table from 'cli-table3';
-import { PreviewCommandLoggerAction as LoggerAction } from 'common/logger-actions';
-import { readAppdata, type SiteData } from 'cli/lib/appdata';
+import { SiteCommandLoggerAction as LoggerAction } from 'common/logger-actions';
+import { getSiteUrl, readAppdata, type SiteData } from 'cli/lib/appdata';
+import { connect, disconnect } from 'cli/lib/pm2-manager';
+import { getColumnWidths } from 'cli/lib/utils';
+import { isServerRunning } from 'cli/lib/wordpress-server-manager';
 import { Logger, LoggerError } from 'cli/logger';
 import { StudioArgv } from 'cli/types';
 
-interface SiteTable {
-	id: string;
+interface SiteListEntry {
+	status: string;
 	name: string;
 	path: string;
+	url: string;
 	phpVersion: string;
 }
 
-function getSitesCliTable( sites: SiteData[] ) {
-	const table = new Table( {
-		head: [ __( 'Name' ), __( 'Path' ), __( 'ID' ), __( 'PHP' ) ],
-		style: {
-			head: [ 'cyan' ],
-			border: [ 'grey' ],
-		},
-		wordWrap: true,
-		wrapOnWordBoundary: false,
-	} );
+function getPrettyPath( path: string ): string {
+	return path.replace( process.cwd(), '.' ).replace( os.homedir(), '~' );
+}
 
-	sites.forEach( ( site ) => {
-		table.push( [ site.name, site.path, site.id, site.phpVersion ] );
-	} );
+async function getSiteListData( sites: SiteData[] ) {
+	const result: SiteListEntry[] = [];
 
-	return table;
-}
+	for await ( const site of sites ) {
+		const isOnline = await isServerRunning( site.id );
```

Comment:
`isServerRunning` will eventually call `listProcesses` I think we could do some optimization here to only fetch the list of running processes once.